File: sources/WCSF2010/Trunk/Source/WebClientFactoryGP/WebClientFactoryPackage/Templates/Projects/Module.VB/FoundationalModuleInitializer.vb
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports Microsoft.Practices.CompositeWeb
Imports Microsoft.Practices.CompositeWeb.Interfaces
Imports Microsoft.Practices.CompositeWeb.Services
Imports Microsoft.Practices.CompositeWeb.Configuration
Imports Microsoft.Practices.CompositeWeb.EnterpriseLibrary.Services


Public Class $ModuleName$ModuleInitializer
    Inherits ModuleInitializer

    Public Overrides Sub Load(ByVal container As CompositionContainer)

        MyBase.Load(container)

        AddGlobalServices(container.Services)
    End Sub

    Protected Overridable Sub AddGlobalServices(ByVal globalServices As IServiceCollection)

        ' TODO: add a service that will be visible to any module
    End Sub

    Public Overrides Sub Configure(ByVal services As IServiceCollection, ByVal moduleConfiguration As System.Configuration.Configuration)

    End Sub

End Class

File: sources/WCSF2010/Trunk/Source/WebClientFactoryGP/WebClientFactoryPackage/bin/Debug/Templates/Projects/Module.CS/Foundational.ModuleInitializer.cs
//===============================================================================
// Microsoft patterns & practices
// Web Client Software Factory 2010
//===============================================================================
// Copyright (c) Microsoft Corporation.  All rights reserved.
// THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY
// OF ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT
// LIMITED TO THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
// FITNESS FOR A PARTICULAR PURPOSE.
//===============================================================================
// The example companies, organizations, products, domain names,
// e-mail addresses, logos, people, places, and events depicted
// herein are fictitious.  No association with any real company,
// organization, product, domain name, email address, logo, person,
// places, or events is intended or should be inferred.
//===============================================================================
using System;
using System.Collections.Generic;
using System.Text;
using Microsoft.Practices.CompositeWeb;
using Microsoft.Practices.CompositeWeb.Interfaces;
using Microsoft.Practices.CompositeWeb.Services;
using Microsoft.Practices.CompositeWeb.Configuration;
using Microsoft.Practices.CompositeWeb.EnterpriseLibrary.Services;

namespace $ModuleNamespace$
{
    public class $ModuleName$ModuleInitializer : ModuleInitializer
	{
        public override void Load(CompositionContainer container)
		{
			base.Load(container);

			AddGlobalServices(container.Services);
		}

        protected virtual void AddGlobalServices(IServiceCollection globalServices)
		{
            // TODO: add a service that will be visible to any module
		}		       

        public override void Configure(IServiceCollection services, System.Configuration.Configuration moduleConfiguration)
        {           
        }
    }
}
